front_end/html/exercise_shared.html

```diff
@@ -205,48 +205,53 @@ <h6>Copy exercise within the same assignment:</h6>
                     var test_output = "<h6>Test outputs:</h6>\n\n";
                         var tests = json_data.tests;
                         {% for i in range(len(exercise_details["tests"]))%}
-                            button_bd_color = (tests[{{i}}].passed ? "passing-bd-color" : "failing-bd-color");
-                            test_message = (tests[{{i}}].passed ? "passed" : "did not pass");
-                            {% if exercise_details["tests"][i]["test_instructions"] == "" %}
-                                test_output += `<div class='${button_bd_color} test-content' style='cursor:pointer;' onclick='showTest(${tests[{{i}}].test})'><h6 style="padding: 1.25em 1.5em;margin:0;">Test {{i + 1}} ${test_message}.</h6>`;
-                                test_output += `<div id="test_${tests[{{i}}].test}" style="padding: 0 1.25em 1.5em 1.25em;"><hr style="margin: 0 0 1.5rem 0;">
-                                  <h6>Test code:</h6>
-                                  <pre>{{exercise_details["tests"][i]["code"]}}</pre>
-                                  <h6>Your output:</h6>\n\n`;
-                                {% if exercise_details["output_type"] == "txt" %}
-                                    if (tests[{{i}}].text_output == "")
-                                        test_output += "<p><pre>[Your solution produced no output.]</pre></p>";
-                                    else
-                                        test_output += `<p><pre>${tests[{{i}}].text_output}</pre><p>`;
-                                    test_output += `<h6>Expected output:</h6>\n\n`;
-                                    test_output += `<pre>{{exercise_details["tests"][i]["text_output"]}}</pre>`;
-                                {% else %}
-                                    if (tests[{{i}}]["image_output"] == "") {
-                                        if (tests[{{i}}]["text_output"] == "")
-                                            test_output += "<p><pre>[Your solution did not produce an image.]</pre></p>";
+                            try {
```

Comment:
This ensures that test output will only be displayed if a submission for each required test is able to be found.

Reply:
Thanks. Please add this comment to the code.